### PR TITLE
fix line numbers when inline sourcemap is used

### PR DESCRIPTION
**Pull request by mstoykov**
## What?

Fix sourcemaps when they are inlined.

Before that if there was an inline sourcemap we won't load it and go through the already used method of updating it. 

This isn't a problem if babel actually transpiles it as it will load it, but in case the sourcecode was already transpiled with inline maps this will now update the lines correctly.

## Why?

To have correct lines when people user sourcemaps . 

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have added tests for my changes.
- [x] I have run linter locally (`make lint`) and all checks pass.
- [x] I have run tests locally (`make tests`) and all tests pass.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

## Related PR(s)/Issue(s)

#3689

### CODE REVIEW

File: js/compiler/compiler.go [hunk 1]
```diff
@@ -210,9 +211,17 @@ func (c *Compiler) compileImpl(
 	state := compilationState{srcMap: srcMap, compiler: c, wrapped: wrap}
 	if wrap {
 		conditionalNewLine := ""
-		if strings.Contains(code, "//# sourceMappingURL=") {
+		if index := strings.LastIndex(code, "//# sourceMappingURL="); index != -1 {
 			// the lines in the sourcemap (if available) will be fixed by increaseMappingsByOne
 			conditionalNewLine = "\n"
+			newCode, err := state.updateInlineSourceMap(code, index)
+			if err != nil {
+				c.logger.Warn("while compiling %q we couldn't update it's inline sourcemap which might lead "+
```
Comment (codebien) [inline js/compiler/compiler.go:219]: It is not clear to me as a user, if I can fix it in some way. Do we have a suggestion? Or the answer is `there isn't much you can do`.

**Review by szkiba — COMMENTED** [review]
In the case of the originally attached script, the line number has been corrected. At the same time, if I transfer the throw expression to another function, it no longer works.

```ts
interface User {
  name: string;
  id: number;
}

class UserAccount implements User {
  name: string;
  id: number;

  constructor(name: string) {
    this.name = name;
    this.id = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
  }
}

function newUser(name: string): User {
  throw "other";
  return new UserAccount(name);
}

export { User, newUser };
```

File: js/compiler/compiler.go (hunk 1 above)
Comment (mstoykov) [reply js/compiler/compiler.go:219]: I would argue this warning is mostly so that when they report an issue - they report it with this warning

**Review by szkiba — DISMISSED** [review]
It works now.

File: js/compiler/compiler.go [hunk 2]
```diff
@@ -210,9 +211,17 @@ func (c *Compiler) compileImpl(
 	state := compilationState{srcMap: srcMap, compiler: c, wrapped: wrap}
 	if wrap {
 		conditionalNewLine := ""
-		if strings.Contains(code, "//# sourceMappingURL=") {
+		if index := strings.LastIndex(code, "//# sourceMappingURL="); index != -1 {
 			// the lines in the sourcemap (if available) will be fixed by increaseMappingsByOne
 			conditionalNewLine = "\n"
+			newCode, err := state.updateInlineSourceMap(code, index)
+			if err != nil {
+				c.logger.Warnf("while compiling %q we couldn't update it's inline sourcemap which might lead "+
```
Comment (codebien) [inline js/compiler/compiler.go:219]:
```suggestion
				c.logger.Warnf("while compiling %q, couldn't update its inline sourcemap which might lead "+
```

**Review by codebien — COMMENTED** [review]
LGTM, just a small text suggestion

**Review by codebien — APPROVED** [review]

**Review by szkiba — APPROVED** [review]
LGTM

**Review by olegbespalov — APPROVED** [review]

**Pull request merged**